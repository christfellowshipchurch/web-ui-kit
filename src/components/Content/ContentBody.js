import React from 'react';
import { Parser } from 'html-to-react';

const htmlToReactParser = new Parser();

const ContentBody = ({ className, children }) =>
  <div className={className}>
    {htmlToReactParser.parse(children)}
  </div>

const defaultProps = {};
const propTypes = {};

ContentBody.defaultProps = defaultProps;
ContentBody.propTypes = propTypes;

export default ContentBody;
