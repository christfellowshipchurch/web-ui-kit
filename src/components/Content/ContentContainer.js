import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {
  contentContainer, contentMedia, contentRight, contentLeft, contentTop, contentBottom
} from '../css/styles.module.css'

import Media from '../Media'

const ContentContainer = ({
  layout, imageUrl, imageAlt, videoUrl, ratio, children, className, rounded, media
}) => {
  const layouts = {
    default: contentBottom,
    inverted: contentTop,
    left: contentLeft,
    right: contentRight,
  };

  const mediaItem = media
    ? (
      <div className={contentMedia}>
        <Media {...media} />
      </div>
    )
    : imageUrl || videoUrl
      ? (
        <div className={contentMedia}>
          <Media ratio={ratio} imageUrl={imageUrl} imageAlt={imageAlt} videoUrl={videoUrl} rounded={rounded} />
        </div>
      )
      : null

  return (
    <div className={classNames(contentContainer, className)}>
      {mediaItem}
      <div className={layouts[layout]}>
        {children}
      </div>
    </div>
  );
};

const defaultProps = {
  layout: 'default',
  imageUrl: null,
  imageAlt: null,
  videoUrl: null,
  ratio: '16by9',
  className: '',
  rounded: false,
  media: null
};

const propTypes = {
  layout: PropTypes.string,
  imageUrl: PropTypes.string,
  imageAlt: PropTypes.string,
  videoUrl: PropTypes.string,
  ratio: PropTypes.string,
  className: PropTypes.string,
  rounded: PropTypes.bool,
  media: PropTypes.object
};

ContentContainer.defaultProps = defaultProps;
ContentContainer.propTypes = propTypes;

export default ContentContainer;
