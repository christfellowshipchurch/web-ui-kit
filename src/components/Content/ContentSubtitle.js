import React from 'react'

const ContentSubtitle = ({ className, children, larger = false }) => (
  larger ?
    <h5 className={className}>
      {children}
    </h5> :
    <h6 className={className}>
      {children}
    </h6>
)

export default ContentSubtitle
