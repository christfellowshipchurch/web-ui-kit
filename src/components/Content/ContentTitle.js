import React from 'react'

const ContentTitle = ({ className, children, larger = false }) => (
  larger ?
    <h1 className={className}>
      {children}
    </h1> :
    <h2 className={className}>
      {children}
    </h2>
)

export default ContentTitle;
