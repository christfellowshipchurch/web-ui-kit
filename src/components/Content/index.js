import Wrapper from './ContentContainer'
import Title from './ContentTitle'
import Subtitle from './ContentSubtitle'
import Body from './ContentBody'

const Content = Wrapper
Content.Title = Title
Content.Subtitle = Subtitle
Content.Body = Body

export default Content