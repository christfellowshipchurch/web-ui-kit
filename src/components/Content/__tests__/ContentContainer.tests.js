import React from 'react'
import Content from '../../Content'
import renderer from 'react-test-renderer'

describe('ContentContainer', () => {
    it('renders content in default layout without media', () => {
        const tree = renderer.create(
            <Content
                ratio="21by9"
                layout="default"
            >
                <Content.Title>Title</Content.Title>
                <Content.Body>Content body.</Content.Body>
            </Content>
        )

        expect(tree).toMatchSnapshot();
    })

    it('renders an image and content in default layout', () => {
        const tree = renderer.create(
            <Content
                imageUrl="https://image.url.com"
                imageAlt="Image Alt"
                ratio="21by9"
                layout="default"
            >
                <Content.Title>Title</Content.Title>
                <Content.Body>Content body.</Content.Body>
            </Content>
        )

        expect(tree).toMatchSnapshot();
    })

    it('renders a video with background image and content in default layout', () => {
        const tree = renderer.create(
            <Content
                videoUrl="https://video.url.com"
                imageUrl="https://image.url.com"
                imageAlt="Image Alt"
                ratio="21by9"
                layout="default"
            >
                <Content.Title>Title</Content.Title>
                <Content.Body>Content body.</Content.Body>
            </Content>
        )

        expect(tree).toMatchSnapshot();
    })

    it('renders a subtitle in the content in the default size', () => {
        const tree = renderer.create(
            <Content
                videoUrl="https://video.url.com"
                imageUrl="https://image.url.com"
                imageAlt="Image Alt"
                ratio="21by9"
                layout="default"
            >
                <Content.Title>Title</Content.Title>
                <Content.Subtitle>Subtitle</Content.Subtitle>
                <Content.Body>Content body.</Content.Body>
            </Content>
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a larger title and subtitle', () => {
        const tree = renderer.create(
            <Content
                videoUrl="https://video.url.com"
                imageUrl="https://image.url.com"
                imageAlt="Image Alt"
                ratio="21by9"
                layout="default"
            >
                <Content.Title larger>Title</Content.Title>
                <Content.Subtitle larger>Subtitle</Content.Subtitle>
                <Content.Body>Content body.</Content.Body>
            </Content>
        )

        expect(tree).toMatchSnapshot()
    })
})