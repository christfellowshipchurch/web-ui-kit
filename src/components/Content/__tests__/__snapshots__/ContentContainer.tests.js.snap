// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer renders a larger title and subtitle 1`] = `
<div
  className=""
>
  <div>
    <div
      className=""
    >
      <div>
        <div />
      </div>
      <video
        autoPlay={true}
        loop={true}
        muted={true}
        playsInline={true}
      >
        <source
          src="https://video.url.com"
          type="video/mp4"
        />
      </video>
    </div>
  </div>
  <div>
    <h1>
      Title
    </h1>
    <h5>
      Subtitle
    </h5>
    <div>
      Content body.
    </div>
  </div>
</div>
`;

exports[`ContentContainer renders a subtitle in the content in the default size 1`] = `
<div
  className=""
>
  <div>
    <div
      className=""
    >
      <div>
        <div />
      </div>
      <video
        autoPlay={true}
        loop={true}
        muted={true}
        playsInline={true}
      >
        <source
          src="https://video.url.com"
          type="video/mp4"
        />
      </video>
    </div>
  </div>
  <div>
    <h2>
      Title
    </h2>
    <h6>
      Subtitle
    </h6>
    <div>
      Content body.
    </div>
  </div>
</div>
`;

exports[`ContentContainer renders a video with background image and content in default layout 1`] = `
<div
  className=""
>
  <div>
    <div
      className=""
    >
      <div>
        <div />
      </div>
      <video
        autoPlay={true}
        loop={true}
        muted={true}
        playsInline={true}
      >
        <source
          src="https://video.url.com"
          type="video/mp4"
        />
      </video>
    </div>
  </div>
  <div>
    <h2>
      Title
    </h2>
    <div>
      Content body.
    </div>
  </div>
</div>
`;

exports[`ContentContainer renders an image and content in default layout 1`] = `
<div
  className=""
>
  <div>
    <div
      className=""
    >
      <div>
        <div />
      </div>
    </div>
  </div>
  <div>
    <h2>
      Title
    </h2>
    <div>
      Content body.
    </div>
  </div>
</div>
`;

exports[`ContentContainer renders content in default layout without media 1`] = `
<div
  className=""
>
  <div>
    <h2>
      Title
    </h2>
    <div>
      Content body.
    </div>
  </div>
</div>
`;
